components/calendar/src/hebrew.rs

```diff
@@ -0,0 +1,1220 @@
+            return Err(CalendarError::UnknownEra(era.0, self.debug_name()));
+        };
+
+        ArithmeticDate::new_from_codes(self, year, month_code, day).map(CivilHebrewDateInner)
+    }
+
+    fn date_from_iso(&self, iso: Date<Iso>) -> Self::DateInner {
+        let fixed_iso = Iso::fixed_from_iso(*iso.inner());
+        Self::civil_hebrew_from_fixed(fixed_iso).inner
+    }
+
+    fn date_to_iso(&self, date: &Self::DateInner) -> Date<Iso> {
+        let fixed_hebrew = Self::fixed_from_civil_hebrew(*date);
+        Iso::iso_from_fixed(fixed_hebrew)
+    }
+
+    fn months_in_year(&self, date: &Self::DateInner) -> u8 {
+        date.0.months_in_year()
+    }
+
+    fn days_in_year(&self, date: &Self::DateInner) -> u16 {
+        date.0.days_in_year()
+    }
+
+    fn days_in_month(&self, date: &Self::DateInner) -> u8 {
+        date.0.days_in_month()
+    }
+
+    fn offset_date(&self, date: &mut Self::DateInner, offset: DateDuration<Self>) {
+        date.0.offset_date(offset)
+    }
+
+    fn until(
+        &self,
+        date1: &Self::DateInner,
+        date2: &Self::DateInner,
+        _calendar2: &Self,
+        _largest_unit: DateDurationUnit,
+        _smallest_unit: DateDurationUnit,
+    ) -> DateDuration<Self> {
+        date1.0.until(date2.0, _largest_unit, _smallest_unit)
+    }
+
+    fn debug_name(&self) -> &'static str {
+        "CivilHebrew"
+    }
+
+    fn year(&self, date: &Self::DateInner) -> types::FormattableYear {
+        Self::year_as_hebrew(date.0.year)
+    }
+
+    fn month(&self, date: &Self::DateInner) -> types::FormattableMonth {
+        date.0.month()
+    }
+
+    fn day_of_month(&self, date: &Self::DateInner) -> types::DayOfMonth {
+        date.0.day_of_month()
+    }
+
+    fn day_of_year_info(&self, date: &Self::DateInner) -> types::DayOfYearInfo {
+        let prev_year = date.0.year.saturating_sub(1);
+        let next_year = date.0.year.saturating_add(1);
+        types::DayOfYearInfo {
+            day_of_year: date.0.day_of_year(),
+            days_in_year: date.0.days_in_year(),
+            prev_year: Self::year_as_hebrew(prev_year),
+            days_in_prev_year: Self::days_in_provided_year(prev_year),
+            next_year: Self::year_as_hebrew(next_year),
+        }
+    }
+}
+
+impl CivilHebrew {
+    /// Constructs a new Civil Hebrew Calendar
+    pub fn new() -> Self {
+        Self
+    }
+
+    // Converts a Biblical Hebrew Date (which considers Tishri the start of the year) to a Civil Hebrew Year (year starts at Nisan).
+    #[allow(clippy::unwrap_used)]
+    fn biblical_to_civil_date(biblical_date: BookHebrew) -> Date<CivilHebrew> {
+        let month = biblical_date.month;
+        let civil_year = biblical_date.year;
+        let mut civil_month;
+
+        if month >= TISHRI {
+            civil_month = (month + 6) % 12;
+            if civil_month == 0 {
+                civil_month = 12;
+            }
+        } else {
+            civil_month = month + 6;
+        }
```

Comment:
I think this can be simplified a little; I don't think you need the special case for `month >= TISHRI`

```suggestion
        civil_month = (month + 6) % 12;
        if civil_month == 0 {
            civil_month = 12;
        }
```